Assignment4/raulmoldes - 54032154/Task07.py

```diff
@@ -0,0 +1,183 @@
+# %% [markdown]
+# **Task 07: Querying RDF(s)**
+
+# %%
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+# %% [markdown]
+# First let's read the RDF file
+
+# %%
+from rdflib import Graph, Namespace, Literal, XSD
+from rdflib.namespace import RDF, RDFS, FOAF
+from rdflib.plugins.sparql import prepareQuery
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0/"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+# %% [markdown]
+# **TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**
+
+# %% [markdown]
+# Using `SPARQL` on RDFLib
+# 
+
+# %%
+# TO DO
+#Creating the namespace object
+ns = Namespace("http://somewhere#")
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0/")
+# Binding the prefixes
+g.bind('rdfs',RDFS)
+g.bind('ns',ns)
+g.bind('foaf',FOAF)
+g.bind('rdf',RDF)
+g.bind('vcard',VCARD)
+
+# Define the SPARQL query
+query_text = """    
+    SELECT ?subclass WHERE {
+        ?subclass rdfs:subClassOf ns:LivingThing .
```

Comment:
missing the subclasses of subclasses of Living Thing (and so on). Hint: make the function recursive